### ISSUE TITLE
[feature] provide pyinstaller hook as entry_point

### ISSUE THREAD

**Issue by samschott** [jaraco/keyring#471]
After longstanding issues with out-of-date or missing pyinstaller hooks, PyInstaller 4.0 allows packages to provide hooks as an entry point `pyinstaller40`. This is explained quite well in their [documentation](https://pyinstaller.readthedocs.io/en/stable/hooks.html#providing-pyinstaller-hooks-with-your-package).

One of the main issues with PyInstaller is the lack of package metadata and therefore the inability to discover entry points, as described in #324. AFAIK, this is a deliberate choice by PyInstaller, possibly to reduce app sizes, and if a package relies on metadata to be available at runtime it should be explicitly provided with a hook using `PyInstaller.utils.hooks.copy_metadata`.

What do you think of providing such a hook with keyring as an entry_point? It will then be discovered and used by PyInstaller automatically. If you think that this is a good idea, I am happy to submit a PR.

**Comment by samschott** [comment]
This could probably also replace the currently provided `hook-keyring.backend.py` by using `PyInstaller.utils.hooks.collect_submodules` to add backends to hidden imports.

**Comment by jaraco** [comment]
Yes, that sounds good to me. Would you be willing to implement it and validate the approach?
